app/precompiled.py

```diff
@@ -628,6 +631,23 @@ def is_notify_tag_present(pdf):
     ) == 'NOTIFY'
 
 
+def redact_precompiled_letter_address_block(pdf, address_regex):
+    options = pdf_redactor.RedactorOptions()
+
+    options.content_filters = []
+    options.content_filters.append((
+        re.compile(address_regex),
+        lambda m: " "
+    ))
+    options.input_stream = pdf
+    options.output_stream = BytesIO()
+
+    pdf_redactor.redactor(options)
+
+    options.output_stream.seek(0)
+    return options.output_stream.read()
```

Comment:
do you think we'll ever want greater control over what happens when we _don't_ redact text?

currently we log, but there's no way for this function to know if the address is still there for example.

Reply:
>there's no way for this function to know if the address is still there for example

I'm not sure I understand. If we find no matches or multiple matches we are pretty confident that the function does not change anything right? So the old address would still be there, then covered by the white rectangle, then the formatted block would be put in front of it, no?

Reply:
ah sorry, i meant more that the `redact_precompiled_letter_address_block` that calls the `pdf_redactor.redact` function doesn't know what happened.

For now that's fine because we're just accepting that they will have a doubled up (but covered) address block and will be unsorted. but if we want to change this or do anything else in the future, then it's going to be hard to do anything because we don't report anywhere whether the redaction process worked or not. All we'll have is a context-less log line.

It'd be nice if `pdf_redactor.redact` returned a boolean success flag or something, I think, even if we don't do anything with that for now.

Reply:
ahh, ok, thanks, that makes lots of sense, I will add it 👍